using_restricted_array.rb

```diff
@@ -6,42 +6,93 @@
 
 # Calculates the length of the restricted array. All values are integers.
 # The restricted_array is terminated by 'nil' i.e. array[length] = nil
+
 def length(array)
-  puts "NOT IMPLEMENTED"
+  i = 0
+  until array[i] == "nil"
+    i += 1
+  end
+  return i
 end
 
-# Prints each integer values in the array
+#Prints each integer values in the array
 def print_array(array)
-  puts "NOT IMPLEMENTED"
+  i = 0
+  until array[i] == "nil"
+    i += 1
+    print array[i]
+  end
+  return i
+
 end
 
 # For an unsorted array, searches for 'value_to_find'.
 # Returns true if found, false otherwise.
 def search(array, length, value_to_find)
-  puts "NOT IMPLEMENTED"
+  i = 0
+  found = false
+  until array[i] == value_to_find
```

Comment:
This should be `until i == length`. You already check to see if array[i] is the value_to_find on line 36. In the case where the value_to_find is not in the array, you may go out of bounds in the array in the code as it currently stands.